### PR TITLE
modify roll test=develop

### PR DESCRIPTION
**Pull request by yaoxuefeng6**
### PR types
 Others

### PR changes
 APIs

### Describe
 
According to paddle 2.0 standard
1, change attr name 'dim' to 'axis'
2, change api input name 'input' to 'x'
3, add check axis value in python code and related ut.
4,change example code to imperative mode

### CODE REVIEW

File: python/paddle/tensor/manipulation.py [hunk 1]
```diff
@@ -104,7 +104,7 @@ def flip(input, dims, name=None):
     return out
 
 
-def roll(input, shifts, dims=None):
+def roll(input, shifts, axis=None, name=None):
     """
```
Comment (jzhang533) [inline python/paddle/tensor/manipulation.py:108]: it should be : paddle.tensor.roll(x, shifts, axis=None, name=None)
according to the latest argument convention.

File: python/paddle/tensor/manipulation.py [hunk 2]
```diff
@@ -117,7 +117,7 @@ def roll(input, shifts, dims=None):
         input (Variable): The input tensor variable.
         shifts (int|list|tuple): The number of places by which the elements
                            of the `input` tensor are shifted.
-        dims (int|list|tuple|None): Dimentions along which to roll.
+        axis (int|list|tuple|None): Dimentions along which to roll.
```
Comment (jzhang533) [inline python/paddle/tensor/manipulation.py]: Dimensions -> axis
上面的简介部分：``Roll the input tensor along ...'' 需要换一种说法, character level copy是不可以的。
示例代码中用`paddle.enable_imperative()`开启动态图模式，方便未来默认动态图的时候统一调整示例代码。

File: python/paddle/tensor/manipulation.py (hunk 1 above)
Comment (yaoxuefeng6) [reply python/paddle/tensor/manipulation.py:108]: ok

File: paddle/fluid/operators/roll_op.h [hunk 3]
```diff
@@ -94,8 +94,8 @@ class RollKernel : public framework::OpKernel<T> {
       PADDLE_ENFORCE_EQ(
           dims[i] < input_dim.size() && dims[i] >= (0 - input_dim.size()), true,
           platform::errors::OutOfRange(
-              "Attr(dims[%d]) is out of range, It's expected "
-              "to be in range of [-%d, %d]. But received Attr(dims[%d]) = %d.",
+              "Attr(axis[%d]) is out of range, It's expected "
+              "to be in range of [-%d, %d]. But received Attr(axis[%d]) = %d.",
```
Comment (jzhang533) [inline paddle/fluid/operators/roll_op.h:98]: Python端也增加对axis范围的检查，防止打印出来call stack。

**Review by 123malin — DISMISSED** [review]
LGTM

**Review by jzhang533 — APPROVED** [review]
LGTM

**Review by XiaoguangHu01 — APPROVED** [review]
LGTM

**Review by liym27 — APPROVED** [review]
LGTM for the changes of Attrs of roll op, but please node that it will cause that Paddle inference library of 2.0 version to load model trained by the old version failed.

**Pull request merged**